### PR TITLE
Feature/entity merge

### PR DESCRIPTION
**Pull request by ivanmrsulja**
## Description

Added support for entity merging and splitting. Subtype merge/split is done using merge service and metadata merge is done using existing update methods. For metadata subtypes, regarding person entity, additionl update methods are added to PersonService.

## How should this be tested?

Everything is covered with unit and integration tests (where applicable). However, it is best advides to test this in conjuction with paired [frontend PR](https://github.com/sci2zero/TeslaRIS-frontend/pull/22).

## Checklist

-   [x] I have tested these changes locally
-   [x] I have added tests that cover these changes (if applicable)
-   [x] I have updated the documentation (if applicable)
-   [x] I have added or updated the necessary comments in the code

### CODE REVIEW

File: src/main/java/rs/teslaris/core/controller/MergeController.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    @PatchMapping("/proceedings/{sourceProceedingsId}/target/{targetProceedingsId}")
+    @PreAuthorize("hasAuthority('MERGE_PROCEEDINGS_PUBLICATIONS')")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void switchAllProceedingsPublicationsToOtherProceedings(
+        @PathVariable Integer sourceProceedingsId,
+        @PathVariable Integer targetProceedingsId) {
+        mergeService.switchAllPublicationsToOtherProceedings(sourceProceedingsId,
+            targetProceedingsId);
+    }
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/controller/MergeController.java:104]: Why switchInvolvementsToOtherPerson is not a method in this class?

File: src/main/java/rs/teslaris/core/controller/PrizeController.java [hunk 2]
```diff
@@ -84,4 +85,13 @@ public void deletePrizeProof(@PathVariable Integer prizeId,
                                  @PathVariable Integer proofId) {
         prizeService.deleteProof(proofId, prizeId);
     }
+
+    @PatchMapping("/merge/person/source/{sourcePersonId}/target/{targetPersonId}")
+    @PreAuthorize("hasAuthority('MERGE_PERSON_METADATA')")
+    public void switchInvolvementsToOtherPerson(@PathVariable Integer sourcePersonId,
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/controller/PrizeController.java:91]: Again, why this is not part of MergeController? Moreover, the name of the method should be corrected to switchPrizesToOtherPerson

**Review by chenejac — CHANGES_REQUESTED** [review]
@ivanmrsulja please check my two comments. Great contribution! No complaints for frontend part.

File: src/main/java/rs/teslaris/core/controller/MergeController.java (hunk 1 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/controller/MergeController.java:104]: I don't know, I wanted to discuss the placement of these methods on the meeting but I forgot :D Should I switch them to MergeController? Should Service methods also be switched to MergeService?

File: src/main/java/rs/teslaris/core/controller/PrizeController.java (hunk 2 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/controller/PrizeController.java:91]: Save answer as above :smile:

File: src/main/java/rs/teslaris/core/controller/MergeController.java (hunk 1 above)
Comment (chenejac) [reply src/main/java/rs/teslaris/core/controller/MergeController.java:104]: I would suggest to put them all together into MergeController (and also to use MergeService for all service methods of this type)

File: src/main/java/rs/teslaris/core/controller/PrizeController.java (hunk 2 above)
Comment (chenejac) [reply src/main/java/rs/teslaris/core/controller/PrizeController.java:91]: ok, probably we will move this (and don't forget to rename this one)

File: src/main/java/rs/teslaris/core/controller/MergeController.java (hunk 1 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/controller/MergeController.java:106]: Refactored :smile:

File: src/main/java/rs/teslaris/core/controller/PrizeController.java (hunk 2 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/controller/PrizeController.java]: Renamed and refactored :smile:

**Review by chenejac — APPROVED** [review]

**Pull request merged**